File: samples/replicatedentity-counter/src/main/java/com/example/counter/domain/CounterHandler.java
// FIXME codegen for ReplicatedEntity

package com.example.counter.domain;

import com.akkaserverless.javasdk.impl.replicatedentity.ReplicatedEntityHandler;
import com.akkaserverless.javasdk.replicatedentity.CommandContext;
import com.akkaserverless.javasdk.replicatedentity.ReplicatedCounter;
import com.akkaserverless.javasdk.replicatedentity.ReplicatedEntity;
import com.example.counter.CounterApi;
import com.google.protobuf.Empty;

public class CounterHandler extends ReplicatedEntityHandler<ReplicatedCounter, Counter> {

  public CounterHandler(Counter entity) {
    super(entity);
  }

  @Override
  public ReplicatedEntity.Effect<?> handleCommand(
      String commandName, ReplicatedCounter data, Object command, CommandContext context) {
    switch (commandName) {

      case "Increase":
        return entity().increase(data, (CounterApi.IncreaseValue) command);

      case "Decrease":
        return entity().decrease(data, (CounterApi.DecreaseValue) command);

      case "GetCurrentCounter":
        return entity().getCurrentCounter(data, (CounterApi.GetCounter) command);

      default:
        throw new ReplicatedEntityHandler.CommandHandlerNotFound(commandName);
    }
  }
}

File: samples/scala-replicated-examples/src/main/scala/com/example/replicated/counter/CounterHandler.scala
package com.example.replicated.counter

import com.akkaserverless.replicatedentity.ReplicatedData
import com.akkaserverless.scalasdk.impl.replicatedentity.ReplicatedEntityHandler
import com.akkaserverless.scalasdk.replicatedentity.CommandContext
import com.akkaserverless.scalasdk.replicatedentity.ReplicatedCounter
import com.akkaserverless.scalasdk.replicatedentity.ReplicatedEntity

class CounterHandler(entity: Counter) extends ReplicatedEntityHandler[ReplicatedCounter, Counter](entity) {

  override def handleCommand(
      commandName: String,
      data: ReplicatedData,
      command: Any,
      context: CommandContext): ReplicatedEntity.Effect[_] = {

    val scalaData = data.asInstanceOf[ReplicatedCounter]
    commandName match {
      case "Increase" =>
        entity.increase(scalaData, command.asInstanceOf[IncreaseValue])
      case "Decrease" =>
        entity.decrease(scalaData, command.asInstanceOf[DecreaseValue])
      case "Get" =>
        entity.get(scalaData, command.asInstanceOf[GetValue])
    }
  }
}
